modules/i18n/input-method/fcitx5.nix

```diff
@@ -0,0 +1,42 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  im = config.i18n.inputMethod;
+  cfg = im.fcitx5;
+  fcitx5Package = pkgs.fcitx5-with-addons.override { inherit (cfg) addons; };
+in {
+  options = {
+    i18n.inputMethod.fcitx5 = {
+      addons = mkOption {
+        type = with types; listOf package;
+        default = [ ];
+        example = literalExample "with pkgs; [ fcitx5-rime ]";
+        description = ''
+          Enabled Fcitx5 addons.
+        '';
+      };
+    };
+  };
+
+  config = mkIf (im.enabled == "fcitx5") {
+    i18n.inputMethod.package = fcitx5Package;
+
+    home.sessionVariables = {
+      GTK_IM_MODULE = "fcitx";
+      QT_IM_MODULE = "fcitx";
+      XMODIFIERS = "@im=fcitx";
+    };
+
+    systemd.user.services.fcitx5-daemon = {
+      Unit = {
+        Description = "Fcitx5 input method editor";
+        PartOf = [ "graphical-session.target" ];
+      };
+      Service.ExecStart = "${fcitx5Package}/bin/fcitx5";
+      Install.WantedBy = [ "graphical-session.target" ];
+    };
+  };
```

Comment:
This may be removed in the future. You can check NixOS/nixpkgs#127367 for details. The autostart can be done by systemd with xdg.